packages/ckeditor5-html-support/src/integrations/image.js

```diff
@@ -0,0 +1,222 @@
+/**
+ * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module html-support/integrations/table
+ */
+
+import { Plugin } from 'ckeditor5/src/core';
+import { setViewAttributes } from '../conversionutils.js';
+
+import DataFilter from '../datafilter';
+
+/**
+ * Provides the General HTML Support integration with the {@link module:image/image~Image Image} feature.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class ImageElementSupport extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ DataFilter ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+
+		if ( !( editor.plugins.has( 'ImageInlineEditing' ) && editor.plugins.has( 'ImageBlockEditing' ) ) ) {
+			return;
+		}
+
+		const schema = editor.model.schema;
+		const conversion = editor.conversion;
+		const dataFilter = editor.plugins.get( DataFilter );
+
+		dataFilter.on( 'register:img', ( evt, definition ) => {
+			/* istanbul ignore next */
```

Comment:
As discussed. Consider adding tests instead of ignoring.

Reply:
Removed the condition entirely, as it happened to be outdated.